guides/tutorials/mediation.md

```diff
@@ -0,0 +1,97 @@
+
+:::
+
+A practical example where the use of mediators is beneficial can be found in a mobile environment. A mobile agent, aka. your mobile wallet, receives messages from other agents. Technically, sending HTTP messages to a mobile agent is impossible because the agent does not have a public IP. As you have probably seen in previous examples, a workaround for this scenario is to use ngrok. Another option is to use a mediator. When an agent is configured to use a mediator, messages to that agent are routed through the mediator and from there to its destination. Thus, a mobile agent can use a "public" mediator to receive messages on its behalf and poll them from time to time.
+
+This is just one example of how a mediator can be used. Further resources about mediators can be found [here](./mediation.md#useful-resources).
+
+:::tip
+
+With mediators, WebSockets come in handy. In a production environment, it is recommendable to register a `WsInboundTransport()` for Alice and a `WsOutboundTransport()` for the mediator. These WebSocket transports allow the mediator to push messages to Alice instead of having Alice pull messages from the mediator repeatedly.
+:::
+
+### 1. Mediator setup
+
+The first part shows how to set up a mediator agent. Generally, there are two ways to do that: deploying it from scratch or using a third-party mediator. This section focuses on the former approach, setting up the mediator from scratch.
+
+The mediator is set up on port 3001 and creates an Out-of-Band invitation that allows multiple other agents to connect to the mediator. Configuring an agent as a mediator demands utilizing the `MediatorModule`. In this example, the `autoAcceptMediationRequests` flag is set to true to accept mediation requests automatically.
+
+```typescript showLineNumbers mediator-setup.ts section-1
+
+```
+
+### 2. Recipient (Alice) setup
+
+The second part shows how to use a mediator. Becoming a recipient of a mediator requires using the `RecipientModule`. The setup is the same whether you use your own or a third-party mediator.
+
+The easiest way to set everything up is by configuring the `RecipientModule` to connect to the mediator on agent startup. This can be done by constructing the module with the `mediatorInvitationUrl`. You can get the mediatorInvitationUrl either from [part one](./mediation.md#1-mediator-setup) by running a mediator yourself or from a public mediator such as the animo-mediator. The following snippet uses the latter approach.
```

Comment:
```suggestion
The easiest way to set everything up is by configuring the `RecipientModule` to connect to the mediator on agent startup. This can be done by constructing the module with the `mediatorInvitationUrl`. You can get the mediatorInvitationUrl either from [part one](./mediation.md#1-mediator-setup) by running a mediator yourself or from a public mediator such as the Animo mediator. The following snippet uses the latter approach.
```